src/main/scala/com/gu/sfl/lib/Merge.scala

```diff
@@ -0,0 +1,36 @@
+package com.gu.sfl.lib
+
+import scala.annotation.tailrec
+import scala.reflect.internal.util.Collections.distinctBy
+
+//This is a virtual straight rob from identity
+object Merge {
+
+  implicit class RichList[A](val l: List[A]) {
+    def tailOrNil = if(l.isEmpty) Nil else l.tail
+  }
+
+  def mergeListBy[A, K](before: List[A], after: List[A])(key: A => K)(implicit ordering: Ordering[A]): List[A] = {
```

Comment:
Is this simpler?
`distinctBy(before ++ after)(key).sorted(ordering)`

Reply:
It looks simpler, admittedly, but it breaks all the tests.

Reply:
See review commended added to tests.
Perhaps merge isn't the intended behaviour and the tests need revising or this isn't a "merge" - could the function be renamed if this isn't a straight forward merge?